### ISSUE TITLE
DiamondOperatorForVariableDefinitionCheck should discard inner classes violations

### ISSUE THREAD

**Issue by stephanenicolas** [sevntu-checkstyle/sevntu.checkstyle#344]
When used in anonymous inner classes, the `<>` operator can't be used, but checkstyle reports this as an error. 

Example : 
```
final List<String> aListOfStrings = new ArrayList<String>() {{
  add("a");
  add("b");
}};
```

Of course, this code is not very elegant, and a workaround is to make the class a proper inner class or to split the list filling and declaration, but still should be ignored.

**Comment by romani** [comment]
Please provide error message  and config, to make sure that we speak about the same Check

**Comment by romani** [comment]
fixed in #270, will be released in 1.13 in several days.
